generate_analysis_graphs.py

```diff
@@ -0,0 +1,106 @@
+import argparse
+from collections import OrderedDict
+
+import altair as alt
+from core_data_modules.logging import Logger
+from core_data_modules.traced_data.io import TracedDataJsonIO
+
+from src.lib import PipelineConfiguration
+
+Logger.set_project_name("ADSS")
+log = Logger(__name__)
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Generates graphs for analysis")
+
+    parser.add_argument("user", help="User launching this program")
+    parser.add_argument("messages_json_input_path", metavar="messages-json-input-path",
+                        help="Path to a JSON file to read the TracedData of the messages data from")
+    parser.add_argument("individuals_json_input_path", metavar="individuals-json-input-path",
+                        help="Path to a JSON file to read the TracedData of the messages data from")
+    parser.add_argument("output_dir", metavar="output-dir",
+                        help="Directory to write the output graphs to")
+
+    args = parser.parse_args()
+
+    user = args.user
+    messages_json_input_path = args.messages_json_input_path
+    individuals_json_input_path = args.individuals_json_input_path
+    output_dir = args.output_dir
+
+    # Read the messages dataset
+    log.info(f"Loading the messages dataset from {messages_json_input_path}...")
+    with open(messages_json_input_path) as f:
+        messages = TracedDataJsonIO.import_json_to_traced_data_iterable(f)
+    log.info(f"Loaded {len(messages)} messages")
+
+    # Read the individuals dataset
+    log.info(f"Loading the individuals dataset from {individuals_json_input_path}...")
+    with open(individuals_json_input_path) as f:
+        individuals = TracedDataJsonIO.import_json_to_traced_data_iterable(f)
+    log.info(f"Loaded {len(individuals)} individuals")
+
+    # Compute the number of messages in each show and graph
```

Comment:
I wonder if it might improve approachability to move this part - that processes the data into a separate file that gets called - so someone looking at code generating the chart doesn't need to worry about all the arg-parsing, and tracedata stuff?

No need to address in this PR, just a thought.

Reply:
Possibly. In any case, it's super easy to do when we need it so will wait until we need to.